src/components/input/index.tsx

```diff
@@ -78,16 +130,35 @@ const Input = forwardRef<HTMLInputElement, Props>(
       setInternalUIValue(targetValue);
       setValueDebounced(targetValue);
     };
-    const onChangeHandler = debounce ? debouncedOnChangeHandler : onChange;
+    const defaultOnChangeHandler: ChangeEventHandler<HTMLInputElement> = (
+      e,
+    ) => {
+      onChange && onChange(e);
+      setInternalUIValue(e.target.value);
+    };
+
+    const onChangeHandler = debounce
+      ? debouncedOnChangeHandler
+      : defaultOnChangeHandler;
 
     return (
-      <ChakraInput
-        {...props}
-        type={type}
-        value={inputValue}
-        onChange={onChangeHandler}
-        ref={ref}
-      />
+      <InputWrapper size={props.size} shouldWrap={!!Icon || isClearable}>
```

Comment:
Do you think we could leverage what chakra provides https://chakra-ui.com/docs/components/input#add-elements-inside-input with those left/right elements?

Reply:
That's exactly what I'm doing. I just don't want add a `<div>` around every input